state-sync/inter-component/consensus-notifications/src/lib.rs

```diff
@@ -0,0 +1,424 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+use std::time::Duration;
+
+use async_trait::async_trait;
+use diem_types::{
+    contract_event::ContractEvent, ledger_info::LedgerInfoWithSignatures, transaction::Transaction,
+};
+use futures::{
+    channel::{mpsc, oneshot},
+    SinkExt,
+};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tokio::time::timeout;
+
+#[derive(Clone, Debug, Deserialize, Error, PartialEq, Serialize)]
+pub enum Error {
+    #[error("Notification failed: {0}")]
+    NotificationError(String),
+    #[error("Hit the timeout waiting for state sync to respond to the notification!")]
+    TimeoutWaitingForStateSync,
+    #[error("Unexpected error encountered: {0}")]
+    UnexpectedErrorEncountered(String),
+}
+
+/// The interface between state sync and consensus, allowing consensus to send
+/// synchronization notifications to state sync.
+#[async_trait]
+pub trait ConsensusNotificationSender: Send + Sync {
+    /// Notify state sync of newly committed transactions and reconfiguration events.
+    async fn notify_new_commit(
+        &self,
+        transactions: Vec<Transaction>,
+        reconfiguration_events: Vec<ContractEvent>,
+    ) -> Result<(), Error>;
+
+    /// Notify state sync to synchronize storage to the specified target.
+    async fn sync_to_target(&self, target: LedgerInfoWithSignatures) -> Result<(), Error>;
```

Comment:
I just want to say, I really appreciate these new names.  `commit()` didn't make sense, since it wasn't committing anything!